tests/foreman/longrun/test_oscap.py

```diff
@@ -268,10 +264,7 @@ def test_positive_upload_to_satellite(self):
                     }
                 )
 
-                # Run "puppet agent -t" twice so that it detects it's,
-                # satellite6 and fetch katello SSL certs.
-                for _ in range(2):
-                    vm.run('puppet agent -t 2> /dev/null')
+                vm.configure_puppet(value['rhel_repo'])
```

Comment:
I wonder how and why `puppet agent run` replaced with `configure_puppet`.

Reply:
Because I need to configure puppet first. This does per-configuration tasks before executing `puppet agent run` https://github.com/SatelliteQE/robottelo/blob/43d5f4f1b98e2ac0047a1678556380506874a231/robottelo/vm.py#L654

Reply:
I got that, but I was wondering, how does it work earlier. Or is this the new way you are following.

Still, not blocking your PR for this.

Reply:
It didn't work earlier.